File: 108 Convert Sorted Array to Binary Search Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        if(nums == null || nums.length == 0) return null;
        if(nums.length == 1) return new TreeNode(nums[0]);
        return getTreeNode(nums, 0, nums.length - 1);
    }
    public TreeNode getTreeNode(int[] nums, int start, int end) {
        if(start > end) return null;
        int mid = start + (end - start) / 2;
        TreeNode r = new TreeNode(nums[mid]);
        r.left = getTreeNode(nums, start, mid - 1);
        r.right = getTreeNode(nums, mid + 1, end);
        return r;
    }
}

File: 108 Convert Sorted Array to Binary Search Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> TreeNode:
        if not nums: return None
        return self.getNode(nums, 0, len(nums) - 1)
    def getNode(self, nums, start, end):
        if start > end: return None
        mid = start + (end - start) // 2
        r = TreeNode(nums[mid])
        r.left = self.getNode(nums, start, mid - 1)
        r.right = self.getNode(nums, mid + 1, end)
        return r
''' Solution in one function
class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> TreeNode:
        if not nums:
            return None

        mid = len(nums) // 2

        root = TreeNode(nums[mid])
        root.left = self.sortedArrayToBST(nums[:mid])
        root.right = self.sortedArrayToBST(nums[mid+1:])

        return root
'''